nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -728,6 +728,12 @@ private Description checkReturnExpression(
       // check for unboxing
       return returnType.isPrimitive() ? doUnboxingCheck(state, retExpr) : Description.NO_MATCH;
     }
+    if (ASTHelpers.isSameType(returnType, Suppliers.JAVA_LANG_VOID_TYPE.get(state), state)) {
```

Comment:
This is restoring precisely the behavior we had before #586; we only special-cased `Void` for return types.